neurom/fst/tests/test_io.py

```diff
@@ -89,3 +90,22 @@ def test_load_neurons_filenames():
     for nrn, name in zip(pop.neurons, NRN_NAMES):
         nt.assert_true(isinstance(nrn, _io.Neuron))
         nt.assert_equal(nrn.name, name)
+
+SWC_PATH = os.path.join(DATA_ROOT, 'swc', 'ordering')
+SWC_ORD_REF = _io.load_neuron(os.path.join(SWC_PATH, 'sample.swc'))
+
+
+def test_load_neuron_mixed_tree_swc():
+    nrn_mix =  _io.load_neuron(os.path.join(SWC_PATH, 'sample_mixed_tree_sections.swc'))
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix), [5, 3])
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix),
+                          get('number_of_sections_per_neurite', SWC_ORD_REF))
+
+
+def test_load_neuron_section_order_break_swc():
+    nrn_mix =  _io.load_neuron(os.path.join(SWC_PATH, 'sample_disordered.swc'))
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix), [5, 3])
+    nt.assert_items_equal(get('number_of_sections_per_neurite', nrn_mix),
+                          get('number_of_sections_per_neurite', SWC_ORD_REF))
+
```

Comment:
OK, no problem! I just think it would be safer if we have more accurate tests...